cppguide.md

```diff
@@ -0,0 +1,632 @@
+```
+
+Note that certain universally-known abbreviations are OK, such as `i` for an iteration variable and `T` for a template parameter.
+
+For the purposes of the naming rules below, a "word" is anything that you would write in English without internal spaces. This includes abbreviations and acronyms; e.g., for "camel case" or "Pascal case," in which the first letter of each word is capitalized, use a name like `startRpc()`, not `startRPC()`.
+
+Template parameters should follow the naming style for their category: type template parameters should follow the rules for type names, and non-type template parameters should follow the rules for variable names.
+
+### File Names
+
+Filenames should written using [Pascal Case](https://wiki.c2.com/?PascalCase). Follow the convention that your project uses.
+
+Examples of acceptable file names:
+
+```
+UseFullClass.cpp
+LSM6DSOXDriver.cpp
+Motors.cpp
+Motors_test.cpp
+```
+
+C++ files should end in `.cpp` and header files should end in `.h`.
+
+Do not use filenames that already exist in `/usr/include`, such as `db.h`.
+
+In general, make your filenames very specific. For example, use `HttpServerLogs.h` rather than `Logs.h`.
+
+A very common case is to have a pair of files called, e.g., `FooBar.h` and `FooBar.cpp`, defining a class called `FooBar`.
+
+### Type Names
+
+Type names are written using Pascal Case. They start with a capital letter and have a capital letter for each new word, with no underscores: `MyExcitingClass`, `MyExcitingEnum`.
+
+The names of all types — classes, structs, type aliases, enums, and type template parameters — have the same naming convention. Type names should start with a capital letter and have a capital letter for each new word. No underscores. For example:
+
+```cpp
+// classes and structs
+class UrlTable { ...
+class UrlTableTester { ...
+struct UrlTableProperties { ...
+
+// typedefs
+typedef hash_map<UrlTableProperties *, std::string> PropertiesMap;
+
+// using aliases
+using PropertiesMap = hash_map<UrlTableProperties *, std::string>;
+
+// enums
+enum UrlTableErrors { ...
+```
+
+### Variable Names
+
+The names of variables (including function parameters) and data members are all lowercase, with underscores between words. Data members of classes (but not structs) additionally have leading underscores. For instance: `a_local_variable`, `a_struct_data_member`, `_a_class_data_member`.
+
+#### Common Variable names
+
+For example:
+
+```cpp
+std::string table_name;  // OK - lowercase with underscore.
+std::string tableName;   // Bad - mixed case.
+```
+
+A generic variable must have the same name as its type. Reducing the number of terms used reduces complexity. It also helps deduce the type of the variable from its name. Most of the time if this rule does not fit well, it's because the name of the type was not well chosen to begin with.
+
+```cpp
+void setTopic(Topic * topic);
+void connect(Database * database);
+
+Point startingPoint, centerPoint;
+Name loginName;
+```
```

Comment:
`startingPoint`, `centerPoint` and `loginName` are a bit confusing after mentioning that variables are in lowercase.

Reply:
done, thanks for pointing this out! :)